### PR TITLE
[Cart] Present cart discounts

### PR DESCRIPTION
**Pull request by lchrusciel**

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
We now have ``discounts`` and ``applied_promotions``, perhaps we should unify this?

**Comment by lchrusciel** [comment]
`discounts` are cart promotion(doesn't matter if applied per item or on the whole cart) and `applied_promotions` are the catalog promotion applied on an item. Maybe we should go with `cartDiscounts` and `catalogDiscounts`? /cc @klederson

File: src/Factory/CartViewFactory.php [hunk 1]
```diff
@@ -82,6 +69,17 @@ public function create(OrderInterface $cart, $localeCode)
             $cartView->payments[] = $this->paymentViewFactory->create($payment, $localeCode);
         }
 
+        $cartDiscounts = [];
```
Comment (pamil) [inline src/Factory/CartViewFactory.php]: `/** @var AdjustmentView[] $cartDiscounts */` before this line?

File: src/View/CartSummaryView.php [hunk 2]
```diff
@@ -4,60 +4,41 @@
...
-     * @var string
-     */
+    /** @var string */
     public $checkoutState;
 
-    /**
-     * @var array
-     */
-    public $items = array();
+    /** @var array|ItemView[[ */
```
Comment (pamil) [inline src/View/CartSummaryView.php]: `[[` -> `[]`

**Pull request merged**